apps/balancer-tools/src/app/apr/(utils)/calculateRoundAPR.ts

```diff
@@ -16,27 +16,53 @@ export async function calculatePoolStats({
   roundId: string;
   poolId: string;
 }) {
-  // TODO:  BAL-646 aggregate historical pool APR when roundId is not provided
+  // TODO: BAL-646 aggregate historical pool APR when roundId is not provided
   const round = Round.getRoundByNumber(roundId);
   const pool = new Pool(poolId);
 
   const endRoundBlockNumber = await getBlockNumberByTimestamp(
-    pool.gauge?.network,
-    round.endDate,
+    pool.gauge?.network ?? 1,
+    round.activeRound ? round.startDate : round.endDate,
```

Comment:
Shouldn't we get current day? Since start of current is pretty much the end of the last

Reply:
Yup, agree with @Rawallon here -- if we're seeing the current round there's no reason why use the block filter since we could just run the query without that parameter and we'll get the latest indexed result on subgraph.